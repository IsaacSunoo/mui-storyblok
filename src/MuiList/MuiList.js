import React, { createElement, lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import List from '@material-ui/core/List';
import Storyblok from '../utils/Storyblok';

const MuiListItem = lazy(() => import('./components/MuiListItem/MuiListItem'));

const MuiList = ({
  rootClass,
  dense,
  disablePadding,
  width,
  content,
}) => {
  const components = {
    MuiListItem,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass, { width });

  return (
    <List
      className={styles.root}
      dense={dense}
      disablePadding={disablePadding}
    >
      <Suspense fallback={<div />}>
        {content.map((item, index) => createElement(
          components[item.component],
          Object.assign(item, { key: index }),
        ))}
      </Suspense>
    </List>
  );
};

export default MuiList;

MuiList.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** mui prop: true | false */
  disablePadding: PropTypes.bool,
  /** mui prop: true | false */
  dense: PropTypes.bool,
  /** width of list */
  width: PropTypes.string,

  /** MuiListItem */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiList.defaultProps = {
  width: '100%',
  dense: false,
  disablePadding: false,
  rootClass: [],
};
