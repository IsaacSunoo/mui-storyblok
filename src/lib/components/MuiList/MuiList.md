MuiList example:

```js noeditor
import MuiList from './MuiList';
```

```js
import MuiList from './MuiList';

const listItemAvatar = {
    component: 'MuiListItemAvatar',
    content: [{
        component: 'MuiIcon',
        iconName: 'explore',
    }],
};

const listItemIcon = {
    component: 'MuiListItemIcon',
    content: [{
        component: 'MuiIcon',
        iconName: 'history',
    }],
};

const listItemText = {
    component: 'MuiListItemText',
    primary: "primary text",
};

const listItemSecondaryAction = {
    component: 'MuiListItemSecondaryAction',
    content: [{
    component: 'MuiIconButtonDownload',
        herf: "https://miro.medium.com/max/1200/1*mk1-6aYaf_Bes1E3Imhc0A.jpeg",
        fileName: "cool.jpeg",
        icon: [{
            component: 'MuiIcon',
            iconName: 'play_for_work',
        }],
    }],
};

const listItem = {
      component: 'MuiListItem',
      listItemAvatar: [listItemAvatar],
      listItemIcon: [listItemIcon],
      listItemText: [listItemText],
      listItemSecondaryAction: [listItemSecondaryAction],
  };


  const content = [listItem];

  <MuiList
    content={[listItem]}
  />
```