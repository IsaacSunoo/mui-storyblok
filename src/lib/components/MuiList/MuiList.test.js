import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiList from './MuiList';

function setup() {
  const props = {
    content: [{
      component: 'MuiListItem',
      listItemText: [{
        component: 'MuiListItemText',
        primary: 'primary',
      }],
    }],
  };
  const comp = shallow(<MuiList {...props} />);
  return { comp, props };
}

describe('<MuiList />', () => {
  it('renders MuiList', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((
      <MemoryRouter>
        <MuiList {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
