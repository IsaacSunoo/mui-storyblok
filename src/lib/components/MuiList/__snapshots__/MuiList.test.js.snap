// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiList /> snapshot 1`] = `
<ul
  className="MuiList-root makeStyles-root-3 MuiList-padding"
>
  <div />
</ul>
`;
